components/layout/query.rs

```diff
@@ -55,18 +153,38 @@ pub fn process_content_box_request(node: ServoLayoutNode<'_>) -> Option<Rect<Au>
     if rects.is_empty() {
         return None;
     }
-
-    Some(rects.iter().fold(Rect::zero(), |unioned_rect, rect| {
+    let rect_union = rects.iter().fold(Rect::zero(), |unioned_rect, rect| {
         rect.to_untyped().union(&unioned_rect)
-    }))
+    });
+
+    let maybe_transform = helper.layout_node_to_root_transform(node);
+
+    match maybe_transform {
+        Some(transform) => Some(PostCompositeQueryHelper::apply_post_composite_transform(
+            rect_union, transform,
+        )),
+        None => Some(rect_union),
+    }
 }
 
-pub fn process_content_boxes_request(node: ServoLayoutNode<'_>) -> Vec<Rect<Au>> {
-    node.fragments_for_pseudo(None)
+pub(crate) fn process_content_boxes_request(
+    node: ServoLayoutNode<'_>,
+    helper: PostCompositeQueryHelper<'_>,
+) -> Vec<Rect<Au>> {
+    let fragments = node.fragments_for_pseudo(None);
+    let content_boxes = fragments
         .iter()
         .filter_map(Fragment::cumulative_border_box_rect)
-        .map(|rect| rect.to_untyped())
-        .collect()
+        .map(|rect| rect.to_untyped());
+
+    let maybe_transform = helper.layout_node_to_root_transform(node);
```

Comment:
Here and above which will make code flow a lot simpler.


```suggestion
    let Some(transform) = helper.layout_node_to_root_transform(node) else {
        return content_boxes.collect()
    }
```